### DISCUSSION TITLE
Install problems: install.php works but root complains about routes

### DISCUSSION THREAD

**Discussion by agrif** [magicbug/Cloudlog#1693] [Q&A] [answered]
I've set up cloudlog running under nginx and php7-fpm -- this part seems to be working fine, at least for a few other simple php scripts I've tested it with.

When running cloudlog, it redirects me to `http://host/install/` correctly, and the install script appears to work -- it accepts my SQL credentials and generates files under `application/config`. Then it redirects me to the root page, which doesn't work.

It tells me:

~~~~
An Error Was Encountered
Unable to determine what should be displayed. A default route has not been specified in the routing file.
~~~~

I've tried enabling logging in `config.php`, but it simply generates the following few lines on every request;

~~~~
INFO - 2022-10-17 00:27:15 --> Config Class Initialized
INFO - 2022-10-17 00:27:15 --> Hooks Class Initialized
INFO - 2022-10-17 00:27:15 --> UTF-8 Support Enabled
INFO - 2022-10-17 00:27:15 --> Utf8 Class Initialized
INFO - 2022-10-17 00:27:15 --> URI Class Initialized
~~~~

Anybody know what could be going wrong here, or at least point me in a new direction?

**Comment by magicbug** [comment, marked as answer]
Sounds like application/config/config.php hasn't been configured correctly.

**Reply by agrif** [reply]
I've been poking at `config.php` without success -- any specific options you think might be relevant?

In particular, I know 'base_url' is set correctly.

**Reply by agrif** [reply]
Whoops! This was entirely due to me being a bonehead.

Somehow my entire `application/config/` directory was empty, except for `config.php` and `database.php` as generated by the installer. Turns out, when half the program is missing, the program doesn't work right!

Restoring those files fixed it. My apologies for the noise.
